python/ray/train/_internal/session.py

```diff
@@ -325,6 +319,25 @@ def get_next(self) -> Optional[_TrainingResult]:
         # Return None if there are no more results to fetch.
         return result
 
+    def _get_result_from_queues(self, block: bool) -> Optional[_TrainingResult]:
+        """Get result from result queue. Pass result from training actor result queue if needed."""
+        result = None
+        try:
+            training_actor_result = self.training_actor_result_queue.get(
+                block=block, timeout=_RESULT_FETCH_TIMEOUT
+            )
+            if training_actor_result:
+                # Must release continue_lock to allow report to work.
+                self.continue_lock.release()
+                self.report(training_actor_result)
```

Comment:
Just return the `training_actor_result: _TrainingResult` so that you go through the same codepath (no need for the lock release / report() call).
There should not be anything on the local result queue.

Reply:
Just wanted to doublecheck with you that doing this is safe. The old `TuneReportCallback` went through the full `tune.report` path so I wanted to do the same here. That path also does stuff like `_auto_fill_metrics` (https://github.com/ray-project/ray/blob/master/python/ray/train/_internal/session.py#L422) but I'm not sure if that's still relevant for us.

Reply:
In this case, `training_actor_result` is just a `Dict`, not a training result. Let's update to reflect that in the typehint and variable name.

Reply:
done.